addons/web/static/src/legacy/js/views/basic/basic_model.js

```diff
@@ -2509,11 +2509,19 @@ var BasicModel = AbstractModel.extend({
                 resourceRef.hasChanged = false;
                 return Promise.resolve(resourceRef);
             } else {
-                const result = await this._rpc({
-                    model: 'ir.model',
-                    method: 'read',
-                    args: [modelId, ['id', 'model']],
-                });
+                let result;
+                if ('default_model_id' in record.context && 'default_model_name' in record.context) {
```

Comment:
This doesn't work if the modelField is updated in the view, as the value in the context won't be re-computed.

Reply:
Couldn't we prevent the value from being changed and use the provided default value IFF the user doesn't have the required access rights?

Reply:
We could, but does this make sense? This already looks like a hack in the framework to handle a specific usecase. This would only add more complexity to it. Not to mention that it would mean that a field (reference) forces another one to be readonly. There won't be any elegant way to do that. We discussed with Martin (mat) and Thomas, he'll go for the initial solution (python side, in documents) instead.